### PR TITLE
(New) Example client cpp qt dbus

### PR DESCRIPTION
**Pull request by witwicki**
(from Frank and Stefan)

We've edited according to comments from Stéphane, and also made some additional changes of our own to simplify integration of this library into any new projects.  At present, it works almost exactly like the Python module.  Feel free to try it out (though noting the dependency on QT5 and C++11).

Best,

-Stefan

### CODE REVIEW

**Comment by witwicki** [comment]
Update: We are QT4 compatible.  I've added a CMakefiles.txt that checks for C++11 compatibility before trying to build this example interface.  Please take a look, try it out, and let me know if you have in mind a better solution.

File: examples/clients/cpp-qt-dbus/main.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    cout << "\n-sending stop event-" << endl;
+    Values event_args({});
+    ThymioIIInterface->sendEventName("Stop",event_args);
+
+    // flag an envent to be listened to, and connect its occurence with your own callback funtion
+    cout << "\n-listening for ObstacleDetected events-" << endl;
+    ThymioIIInterface->connectEvent("ObstacleDetected",my_callback);
+
+
+	return a.exec();
```
Comment (stephanemagnenat) [inline examples/clients/cpp-qt-dbus/main.cpp]: weird indentation. Note that the Aseba convention is to use tabs and configure them for 4 spaces.

File: examples/clients/cpp-qt-dbus/main.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    Values sound_index({max_index});
+    ThymioIIInterface->sendEventName("PlaySound",sound_index);
+}
+
+
+int main(int argc, char *argv[])
+{
+    // QT init
+	QCoreApplication a(argc, argv);
```
Comment (stephanemagnenat) [inline examples/clients/cpp-qt-dbus/main.cpp]: weird indentation

File: examples/clients/cpp-qt-dbus/dbusinterface.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		Values getVariable(const QString& node, const QString& variable);
+		void setVariable(const QString& node, const QString& variable,Values& value);
+        void connectEvent(const QString& eventName, std::function<void(Values)> callback);
+
+		void sendEvent(quint16 eventID, Values& value);
+		void sendEventName(const QString& eventName, const Values& value);
+
+
+	public slots:
+        void dispatchEvent(const QDBusMessage& message);
```
Comment (stephanemagnenat) [inline examples/clients/cpp-qt-dbus/dbusinterface.h]: weird indentation

**Comment by stephanemagnenat** [comment]
In general and before testing, this pull request looks good. Please fix indentation using tabs (corresponding to 4 spaces) and {} on single lines. Also, please rebase your commits on the master branch to have a single consistent commit. Then I will do a last test and merge.

**Comment by witwicki** [comment]
OK.  Ready for your final review and testing.

**Comment by stephanemagnenat** [comment]
Merged on my side.

**Comment by mbonani** [comment]
It seems that it breaks the winodws build. The QtDBus module is a Unix-only library.

**Comment by stephanemagnenat** [comment]
Should be fixed with df0bb71.

**Comment by witwicki** [comment]
Thank you Stéphane for taking care of this.

**Pull request closed without merging**